Do not use certain header files. We define bswap_16 ourselves.
Do not plan to send upstream.

diff --git a/media/libstagefright/frameworks/av/media/libstagefright/id3/ID3.cpp b/media/libstagefright/frameworks/av/media/libstagefright/id3/ID3.cpp
--- a/media/libstagefright/frameworks/av/media/libstagefright/id3/ID3.cpp
+++ b/media/libstagefright/frameworks/av/media/libstagefright/id3/ID3.cpp
@@ -20,17 +20,23 @@
 #include <utils/Log.h>
 
 #include "../include/ID3.h"
 
 #include <media/stagefright/foundation/ADebug.h>
 #include <media/stagefright/DataSource.h>
 #include <media/stagefright/Utils.h>
 #include <utils/String8.h>
-#include <byteswap.h>
+//#include <byteswap.h>
+
+#ifndef bswap_16
+# define bswap_16(p) \
+        (((((uint16_t)(p)) & 0x00ff) << 8) | \
+          (((uint16_t)(p))           >> 8));
+#endif
 
 namespace stagefright {
 
 static const size_t kMaxMetadataSize = 3 * 1024 * 1024;
 
 struct MemorySource : public DataSource {
     MemorySource(const uint8_t *data, size_t size)
         : mData(data),
diff --git a/media/libstagefright/moz.build b/media/libstagefright/moz.build
--- a/media/libstagefright/moz.build
+++ b/media/libstagefright/moz.build
@@ -2,17 +2,17 @@
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 DEFINES['ANDROID_SMP'] = 0
 DEFINES['LOG_NDEBUG'] = 1
 
-if CONFIG['OS_TARGET'] != 'WINNT':
+if CONFIG['OS_TARGET'] != 'WINNT' and CONFIG['OS_TARGET'] != 'SunOS':
    DEFINES['_GLIBCXX_OS_DEFINES'] = True
 
 if CONFIG['OS_TARGET'] == 'WINNT':
     if CONFIG['_MSC_VER']:
         DEFINES['ssize_t'] = 'intptr_t'
         DEFINES['off64_t'] = 'int64_t'
         DEFINES['strcasecmp'] = 'stricmp'
         DEFINES['strncasecmp'] = 'strnicmp'
diff --git a/media/libstagefright/system/core/include/cutils/properties.h b/media/libstagefright/system/core/include/cutils/properties.h
--- a/media/libstagefright/system/core/include/cutils/properties.h
+++ b/media/libstagefright/system/core/include/cutils/properties.h
@@ -12,17 +12,17 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
 
 #ifndef __CUTILS_PROPERTIES_H
 #define __CUTILS_PROPERTIES_H
 
-#include <sys/cdefs.h>
+//#include <sys/cdefs.h>
 #include <stddef.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /* System properties are *small* name value pairs managed by the
 ** property service.  If your data doesn't fit in the provided
